### ISSUE TITLE
Deprecation error from symplify/package-builder

### ISSUE THREAD

**Issue by tobyryuk** [nunomaduro/phpinsights#215]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| Library version  | 1.7.0

I'm running Laravel on Alpine and when I'm running `php artisan insights`, I only get this deprecation warning:
```
"Symplify\PackageBuilder\DependencyInjection\CompilerPass\AutowireSinglyImplementedCompilerPass" is deprecated due to unpredictable behavior and causing too many bugs. Use explicit interface autowiring, see https://symfony.com/doc/current/service_container/autowiring.html#working-with-interfaces
```

Haven't been able to figure out why yet as I can't find anything using it, result from `composer why symplify/package-builder`:
```
nunomaduro/phpinsights         v1.7.0  requires  symplify/package-builder (^6.0)
symplify/coding-standard       v6.0.0  requires  symplify/package-builder (^6.0)
symplify/easy-coding-standard  v6.0.0  requires  symplify/package-builder (^6.0)
```

**Comment by xuanquynh** [comment]
It is seem a deprecation from symplify/package-builder core code. Could you provide more information?
- Laravel version
- composer.json and composer.lock files content.

**Comment by tobyryuk** [comment]
Laravel Version: `5.8.8`

composer.json:
```
{
    "name": "laravel/laravel",
    "description": "The Laravel Framework.",
    "keywords": [
        "framework",
        "laravel"
    ],
    "license": "MIT",
    "type": "project",
    "repositories": [
        {
            "type": "path",
            "url": "./nova"
        },
        {
            "type": "path",
            "url": "./nova-components/CacheManager"
        }
    ],
    "require": {
        "php": ">=7.1.3",
        "aws/aws-sdk-php": "^3.91",
        "cocur/slugify": "^3.2",
        "doctrine/dbal": "^2.9",
        "fideloper/proxy": "~4.0",
        "kayt/cache-manager": "*",
        "laravel/framework": "5.8.*",
        "laravel/horizon": "^3.1",
        "laravel/nova": "*",
        "laravel/passport": "^7.2",
        "laravel/telescope": "^2.0",
        "laravel/tinker": "~1.0",
        "league/uri-parser": "^1.4",
        "onelogin/php-saml": "^3.0",
        "predis/predis": "^1.1",
        "sentry/sentry-laravel": "^1.0"
    },
    "require-dev": {
        "barryvdh/laravel-ide-helper": "^2.6",
        "beyondcode/laravel-dump-server": "^1.0",
        "filp/whoops": "~2.0",
        "friendsofphp/php-cs-fixer": "^2.15",
        "fzaninotto/faker": "~1.4",
        "josiasmontag/laravel-redis-mock": "^1.2",
        "mockery/mockery": "~1.0",
        "nunomaduro/collision": "^3.0",
        "nunomaduro/phpinsights": "^1.7",
        "phpunit/phpunit": "^7.5",
        "symfony/thanks": "^1.0"
    },
    "autoload": {
        "classmap": [
            "database/seeds",
            "database/factories"
        ],
        "psr-4": {
            "App\\": "app/"
        },
        "files": [
            "app/helpers.php"
        ]
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "extra": {
        "laravel": {
            "dont-discover": []
        }
    },
    "scripts": {
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "@php artisan key:generate --ansi"
        ],
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover --ansi"
        ]
    },
    "config": {
        "preferred-install": "dist",
        "sort-packages": true,
        "optimize-autoloader": true
    },
    "minimum-stability": "dev",
    "prefer-stable": true
}
```

composer.lock: https://pastebin.com/BVzQ3W5E

**Comment by xuanquynh** [comment]
It exactly comes from symplify/package-builder package code. Have you configured the **error_reporting** setting of php.ini file? You may modify php.ini that contains this like.
```ini
; php will report all errors but ignore deprecation.
error_reporting = E_ALL & ~E_DEPRECATED
```

**Comment by xuanquynh** [comment]
This [line](https://github.com/Symplify/PackageBuilder/blob/14db22e0c9667aeb81873468b4984a490d2183b6/src/DependencyInjection/CompilerPass/AutowireSinglyImplementedCompilerPass.php#L33) produces the deprecation.

**Comment by tobyryuk** [comment]
Changing the `php.ini` file will just sweep the problem under the rug, as you can see, under the `trigger_error` they've added a `sleep`. I've been trying to find out what uses it but with no luck so far.

**Comment by tobyryuk** [comment]
Seems like it's installing `v6.0.0` of `symplify/easy-coding-standard` for me which still uses the deprecated class, not sure why it's not installing the latest but I'll try to upgrade it.

**Comment by tobyryuk** [comment]
Forcing an update of `symplify/easy-coding-standard` fixed it.

**Comment by nunomaduro** [comment]
@tobyryuk What is the version you are using that fixes it? Can bump our `composer.json` symplify/easy-coding-standard to match that?

**Comment by tobyryuk** [comment]
I think it was fixed from `6.0.1` but I'm currently running `6.0.4` with ease!

**Comment by nunomaduro** [comment]
Yeah, just perform a pull request in our repository that bumps as well that dependency to `^6.0.4`.
